### PR TITLE
Add commonly used synonyms, pathName

### PR DESCRIPTION
**Pull request by garyb**
@cryogenian can you take a look?

Figured we may as well add them here rather than reimplementing ourselves in multiple places.

### CODE REVIEW

File: src/Data/Path/Pathy.purs [hunk 1]
```diff
@@ -199,6 +209,9 @@ module Data.Path.Pathy
     (DirIn _ d) -> Just d
     _           -> Nothing
 
+  pathName :: forall b s. AnyPath b s -> Either (Maybe DirName) FileName
+  pathName = bimap dirName fileName
```
Comment (garyb) [inline src/Data/Path/Pathy.purs:213]: Maybe this should just be `name`? Wasn't sure...

Comment (cryogenian) [inline src/Data/Path/Pathy.purs:213]: :+1: for `pathName`

Comment (cryogenian) [inline src/Data/Path/Pathy.purs:213]: :+1: for `pathName`

**Comment by cryogenian** [comment]
:+1:

**Pull request merged**